### PR TITLE
Create dedicated Attrable mixin for attr_ methods

### PR DESCRIPTION
**Pull request by dduugg**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
`attr_rw` was monkey-patched into `Module`, polluting every namespace, despite being used in only three modules. This PR yanks it into a dedicated mixin, `Attrable`. I also moved in `attr_predicate`, bc it seemed like overkill to have a mixin for each. Finally, I stopped including them from `global`, and required them only in the files where they are used.

### CODE REVIEW

**Review by Bo98 — APPROVED** [review]

File: Library/Homebrew/attrable.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      define_method attr do
+        instance_variable_get("@#{attr.to_s.sub(/\?$/, "")}") == true
+      end
+    end
+  end
+
+  sig { params(attrs: Symbol).void }
+  def attr_rw(*attrs)
+    attrs.each do |attr|
+      module_eval <<-EOS, __FILE__, __LINE__+1
```
Comment (MikeMcQuaid) [inline Library/Homebrew/attrable.rb:17]: Can this be done with `define_method`?

**Review by MikeMcQuaid — APPROVED** [review]
Looks good, good idea @dduugg. One chance for a cleanup, if possible (but might not me).

File: Library/Homebrew/attrable.rb (hunk 1 above)
Comment (dduugg) [reply Library/Homebrew/attrable.rb]: Done! (I simplified it as well, taking advantage of `instance_variable_get` returning `nil` for unset ivars.)

**Review by MikeMcQuaid — APPROVED** [review]
Looks good, lovely cleanup, thanks @dduugg!

**Pull request merged**